src/reactive-membrane.ts

```diff
@@ -35,15 +36,20 @@ export interface ObservableMembraneInit {
     valueObserved?: ReactiveMembraneAccessCallback;
     valueDistortion?: ReactiveMembraneDistortionCallback;
     valueIsObservable?: ReactiveMembraneObservableCallback;
+    magicPropertyKey?: PropertyKey;
 }
 
-function createShadowTarget(value: any): ReactiveMembraneShadowTarget {
+function createShadowTarget(value: any, magicPropertyKey: PropertyKey | undefined): ReactiveMembraneShadowTarget {
     let shadowTarget: ReactiveMembraneShadowTarget | undefined = undefined;
     if (isArray(value)) {
         shadowTarget = [];
     } else if (isObject(value)) {
         shadowTarget = {};
     }
+    if (!isUndefined(magicPropertyKey)) {
+        // installing the magic property as non-configurable, non-writable and non-enumerable
+        ObjectDefineProperty(shadowTarget, magicPropertyKey, ObjectCreate(null));
```

Comment:
Have you considered making the property `writable: true` ? That way it could cover more use cases.